csharp/src/Ice/IncomingResponseFrame.cs

```diff
@@ -84,89 +82,100 @@ public void ReadVoidReturnValue(Communicator communicator)
         /// <param name="payload">The frame data as an array segment.</param>
         public IncomingResponseFrame(Protocol protocol, ArraySegment<byte> payload)
         {
+            protocol.CheckSupported();
             Protocol = protocol;
-            byte replyStatus = payload[0];
-            if (replyStatus > 7)
-            {
-                throw new InvalidDataException(
-                    $"received {Protocol.GetName()} response frame with unknown reply status `{replyStatus}'");
-            }
-            ReplyStatus = (ReplyStatus)replyStatus;
             Payload = payload;
-            if (ReplyStatus == ReplyStatus.UserException || ReplyStatus == ReplyStatus.OK)
+
+            if (Protocol == Protocol.Ice1)
```

Comment:
We could consider moving these checks to ReadException and get rid of the ReplyStatus property assignment here.

Reply:
Fixed as suggested.